### ISSUE TITLE
Stack level too deep when using systemd_* resources in a custom resource

### ISSUE THREAD

**Issue by mattlqx** [nathwill/chef-systemd#117]
It looks as though the dynamic evaluation of the blocks within a systemd_* resource doesn't play well with new_resource usage in a custom resource in cookbook version 3.x.

Example code within a custom resource...
```
action(:create) do
  systemd_timer "health_check_#{check_name}" do
    unit do
      description "Runs health check every #{new_resource.frequency_sec} seconds"
    end
    ...
  end
end
```

Will produce something like this snippet of stack trace:
```
...
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `respond_to?'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `block in build_dsl'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `respond_to?'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `block in build_dsl'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `respond_to?'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:41:in `block in build_dsl'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:33:in `respond_to?'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:33:in `block in build_dsl'
/tmp/kitchen/cache/cookbooks/network/resources/cloudwatch_health_metric.rb:47:in `block (3 levels) in class_from_file'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:53:in `instance_eval'
/tmp/kitchen/cache/cookbooks/systemd/libraries/mixin.rb:53:in `block (2 levels) in build_dsl'
/tmp/kitchen/cache/cookbooks/network/resources/cloudwatch_health_metric.rb:46:in `block (2 levels) in class_from_file'
...
```
Where line 47 of the custom resource is the description line.

Setting an intermediary variable outside the resource and using that instead of the new_resource property works.

**Comment by nathwill** [comment]
man, that's wild 😱 . thanks for the report! i'll take a look at this and see if i can find a workaround, as that's obviously less than desirable...

**Comment by mattlqx** [comment]
Thanks for the quick fix 👍🏻

**Comment by nathwill** [comment]
@mattlqx just published to supermarket as 3.1.2, give it a go!
